repos/system_upgrade/common/actors/initramfs/upgradeinitramfsgenerator/libraries/upgradeinitramfsgenerator.py

```diff
@@ -198,9 +200,32 @@ def copy_boot_files(context):
     api.produce(content)
 
 
+@contextlib.contextmanager
+def mount_upgrade_iso_to_root_dir(root_dir, target_iso):
+    if target_iso:
+        # Mountpoint should be always absolute since it is being mounted in different places (scratch,
+        # targetuserspace), so it would not be clear to what is the path relative
+        if not target_iso.mountpoint.startswith('/'):
+            api.current_logger().warn('Target iso mounpoint is not an absolute path.')
+            mountpoint = target_iso.mountpoint
+        else:
+            mountpoint = target_iso.mountpoint[1:]
+        mountpoint_in_root_dir = os.path.join(root_dir, mountpoint)
+    try:
+        if target_iso:
+            if not os.path.isdir(mountpoint_in_root_dir):
+                os.mkdir(mountpoint_in_root_dir)
+            run(['mount', target_iso.path, mountpoint_in_root_dir])
```

Comment:
Am I right in assuming "-o loop" is not needed anymore these days? Looks like "-t iso9660" is not.

Reply:
Yeah, it should not be necessary anymore according to [man mount#loop-device_support](https://man7.org/linux/man-pages/man8/mount.8.html#LOOP-DEVICE_SUPPORT). I guess it will not hurt to include loop option explicitly, so I have added it there.